### DISCUSSION TITLE
Question: How to know when LSP disconnects?

### DISCUSSION THREAD

**Discussion by igbanam** [neoclide/coc.nvim#4897] [Q&A]
I have a language server which I'm connecting to with a custom extension. This language server is connected to some remote instance via SSH. When that SSH connection is broken, ideally I would like the extension to know and fall into some exponential back-off retry strategy to reconnect.

What's the general mechanism for this?

Ideally, there should be some signal the extension should be able to make available to vim for this — for things like updating the UI, for example.

Also, if there's some general documentation on how to write extensions for coc.nvim, please share. Thank you.

**Comment by fannheyward** [comment]
> know when LSP disconnects

I don't think LSP defined any thing for this by default. You can custom this in your server: request from client and server responses  with special content, if no response received in time or request failed to connected, the client can consider the server was down. Check `:h CocRegisterNotification`.

> documentation on how to write extensions for coc.nvim

No official documents for now, here's some resources maybe helpful for you:
- https://github.com/neoclide/coc.nvim/discussions/4894
- `:h coc-api.txt`
